processing/data_collection/gazette/spiders/sc_joinville.py

```diff
@@ -0,0 +1,56 @@
+import re
+import dateparser
+
+from datetime import datetime
+from scrapy import Request, Spider
+from gazette.items import Gazette
+
+class GazetteElement:
+    DATE_CSS = 'span.article-date::text'
+    MUNICIPALITY_ID = '4209102'
+    PDF_URL = 'https://www.joinville.sc.gov.br{}'
+
+    def __init__(self, element):
+        self.element = element
+
+    def date(self):
+        date_str = self.element.css(self.DATE_CSS)[-1].extract()
```

Comment:
It is much easier to read and understand what is happening with a class.

Reply:
@rennerocha

Reply:
Your class is doing the parsing of a response of a page, so the right place is in the `parse` method of the spider. If you really want to use a separate class, create a ItemLoader and use it how it was designed to be used in Scrapy. Here you are just creating an extra layer of abstraction for something very very simple that may be replaced by a dozen lines of code.

Reply:
In this case I don't want the first `article-date` element, but the last one. Didn't find an `extract_last` method. What do you think I should do here?

Reply:
I was going to suggest you to use an _xpath_ to get the span that contains only the `article-date`, because it is the one that you really want. This should do the trick as it search for only the span where the class content is _exactly_ `article-date`:
`self.element.xpath('.//span[@class="article-date"]//text()').extract_first()`

However, I think we could use and a regex and the `.re` to get the content we want:
`self.element.css('span.article-date::text').re('([\d]+)[ |]+([\w]+)[ |]+([\d]+)')`

Reply:
Done! Liked a lot the `.re` solutions :)